pandas/_libs/tslib.pyx

```diff
@@ -142,16 +152,18 @@ def ints_to_pydatetime(ndarray[int64_t] arr, tz=None, freq=None, box=False):
         ndarray[object] result = np.empty(n, dtype=object)
         object (*func_create)(int64_t, pandas_datetimestruct, object, object)
 
-    if box and is_string_object(freq):
-        from pandas.tseries.frequencies import to_offset
-        freq = to_offset(freq)
-
-    if box:
+    if box == "date":
+        func_create = create_date_from_ts
+    elif box == "timestamp":
         func_create = create_timestamp_from_ts
-    else:
+
+        if is_string_object(freq):
+            from pandas.tseries.frequencies import to_offset
+            freq = to_offset(freq)
+    elif box == "datetime":
         func_create = create_datetime_from_ts
 
-    if tz is not None:
+    if tz is not None and box != "date":
```

Comment:
instead of this, put an assert in the ``box == 'date'`` section that ``tz  is None``

Reply:
```
if box == "date":
        assert (tz is not None), "tz should be None when converting to date"

        func_create = create_date_from_ts
```

@jreback Can I do something like this instead? That way the error will be more specific when tz is used with date.

Reply:
yes absolutely. The point is we should *not* be passing a tz with the date routine.